vm/src/dictdatatype.rs

```diff
@@ -79,6 +100,9 @@ impl<T: Clone> Dict<T> {
         key: K,
         value: T,
     ) -> PyResult<()> {
+        if self.indices.len() > 2 * self.size {
```

Comment:
It would be nice to check how it was implemented in cpython and implement a resize policy.
https://github.com/python/cpython/blob/master/Objects/dictobject.c

Reply:
It seems like they have a very different reasons for resizing. E.g. they need to manually resize what we call `entries` (`ma_values` in cpython), C doesn't handle resizing for them, so they do it by calling `dictresize`.  
Overall they do it quite often, so I assume they don't have a problem we're trying to solve (`indices` map is growing big)